sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -0,0 +1,389 @@
+                        break;
+                      case Types.CLOB:
+                        payload = resultSet.getClob(i);
+                        break;
+                      case Types.DATE:
+                        payload = resultSet.getDate(i);
+                        break;
+                      case Types.DECIMAL:
+                        payload = resultSet.getBigDecimal(i);
+                        break;
+                      case Types.DOUBLE:
+                        payload = resultSet.getDouble(i);
+                        break;
+                      case Types.FLOAT:
+                        payload = resultSet.getFloat(i);
+                        break;
+                      case Types.INTEGER:
+                        payload = resultSet.getInt(i);
+                        break;
+                      case Types.LONGNVARCHAR:
+                        payload = resultSet.getString(i);
+                        break;
+                      case Types.LONGVARCHAR:
+                        payload = resultSet.getString(i);
+                        break;
+                      case Types.NCHAR:
+                        payload = resultSet.getNString(i);
+                        break;
+                      case Types.NCLOB:
+                        payload = resultSet.getNClob(i);
+                        break;
+                      case Types.SMALLINT:
+                        payload = resultSet.getInt(i);
+                        break;
+                      case Types.TIME:
+                        payload = resultSet.getTime(i);
+                        break;
+                      case Types.TIMESTAMP:
+                        payload = resultSet.getTimestamp(i);
+                        break;
+                      case Types.TINYINT:
+                        payload = resultSet.getInt(i);
+                        break;
+                      case Types.VARCHAR:
+                        payload = resultSet.getString(i);
+                        break;
+                      default:
+                        payload = resultSet.getObject(i);
+                        break;
+                    }
+                    record.getColumnValues()[i - 1] = payload;
+                  }
+                } catch (Exception e) {
+                  LOGGER.error("Can't map row", e);
+                }
+                return record;
+              }
+            })).setCoder(SerializableCoder.of(JdbcDataRecord.class));
+
+    PAssert.thatSingleton(
+        output.apply("Count All", Count.<JdbcDataRecord>globally()))
+        .isEqualTo(1000L);
+
+    PAssert.that(output
+        .apply("Map Scientist", MapElements.via(
+            new SimpleFunction<JdbcDataRecord, KV<String, Void>>() {
+              public KV<String, Void> apply(JdbcDataRecord input) {
+                // find NAME column id
+                int index = -1;
+                for (int i = 0; i < input.getColumnNames().length; i++) {
```

Comment:
Again: if you make the example simpler by deleting JdbcDataRecord and using a simple POJO, this loop will be unnecessary and you'll just do regular field access on the POJO.